Add dialog to screenshot a proposal

chadbrower0 opened the pull request:
When the proposal is expanded, an extra button is added next to edit and delete buttons, which opens the screenshot dialog.  The dialog offers options to choose which part of the proposal to screenshot (opinions vs reasons).  The dialog also has buttons to copy the screenshot or the URL to clipboard.  (Copy to clipboard is not well tested, because only https and localhost domains have access to clipboard API.)

tkriplean commented:
Again, this is awesome. I can imagine this functionality to be very useful to hosts. I can also imagine it being useful to save a screenshot of a list of proposals. 

A few notes: 
- The edit/delete buttons don't appear for frozen forums, and thus the screenshot functionality doesn't appear (but it is useful for frozen forums)
- The screenshot functionality will only be available for a host. However, it seems like it might be useful for anyone. Not sure though, keeping the clutter down is nice. 
- (I don't entirely love the edit / delete buttons as they exist, so we might want a different UI approach there entirely)
- The screenshot doesn't copy the background color
- It takes a little while for the screenshot to be available in the dialog. In the meantime, a bunch of errors are logged.
- Copying to clipboard only copies the url, not the image (at least for me!). However, I could right-click on the image preview and save it.  
- I'm wondering it a download button instead of copy to clipboard button would make more sense.

<img width="998" alt="errors" src="https://github.com/Considerit/ConsiderIt/assets/707039/f2698493-d986-4be6-be41-cc4b42864602">

chadbrower0 commented:
Added a white background-color fill.

chadbrower0 commented:
I changed the dialog to offer checkboxes that customize which content to include.  And I added logic to filter out content that is always unwanted, like various buttons.

![image](https://github.com/Considerit/ConsiderIt/assets/17461871/35903173-867d-4c74-be4a-5f686d4b7047)

chadbrower0 commented:
Restyled screenshot dialog for more consistency with main webpage.

![image](https://github.com/Considerit/ConsiderIt/assets/17461871/8dad474b-15dc-436b-b0a3-4c545e9b1adb)
